inifinite loops should only be checked in mktexfmt mode to make sure that
double defined format/engine combinations are accepted
Included upstream, written by Norbert Preining
---
 texk/tetex/fmtutil |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: texlive-bin-2009/texk/tetex/fmtutil
===================================================================
--- texlive-bin-2009.orig/texk/tetex/fmtutil	2011-05-24 21:54:12.000000000 +0200
+++ texlive-bin-2009/texk/tetex/fmtutil	2011-05-24 21:55:52.000000000 +0200
@@ -60,7 +60,7 @@
 
 progname=fmtutil
 argv0=$0
-version=20091009.0222
+version=20091223.1047
 
 cnf=fmtutil.cnf   # name of the config file
 export PATH
@@ -701,6 +701,9 @@
   rm -f $fmtfile
 
   # Check for infinite recursion before running the iniTeX:
+  # We do this check only if we are running in mktexfmt mode
+  # otherwise double format definitions will create an infinite loop, too
+  $mktexfmtMode || mktexfmt_loop=
   case :$mktexfmt_loop: in
   *:"$format/$engine":*)
     abort "Infinite recursion detected, giving up!" ;;
